aten/src/ATen/Utils.h

```diff
@@ -127,4 +128,32 @@ inline int64_t prod_intlist(ArrayRef<int64_t> list) {
   return std::accumulate(list.begin(), list.end(), 1ll, std::multiplies<int64_t>());
 }
 
+/*
+* Utility function used in tensor implementations, which
+* supplies the default generator to tensors, if an input generator
+* is not supplied. The input Generator* is also dynamic casted to
+* the backend generator type (CPU/CUDAGenerator etc.)
+*/
+template <typename T>
+static inline T * check_generator_with_default(Generator * expr, Generator * defaultValue) {
+  if (!expr)
+    expr = defaultValue;
+  T temp_gen;
+  if (temp_gen.getDevice().type() == expr->getDevice().type())
```

Comment:
it seems more direct to add a static method to all subbclasses of Generator and call that, so you don't have to make a fake generator.

Reply:
(Since generators have a lot of fields and state and you don't want to be unnecessarily allocating them)